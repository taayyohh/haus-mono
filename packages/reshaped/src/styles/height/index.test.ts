import getHeightStyles from "./index";

describe("Styles/Height", () => {
	test("handles px value", () => {
		expect(getHeightStyles("50px")).toMatchSnapshot();
	});

	test("handles % value", () => {
		expect(getHeightStyles("50%")).toMatchSnapshot();
	});

	test("handles unit value", () => {
		expect(getHeightStyles(5)).toMatchSnapshot();
	});

	test("handles undefined value", () => {
		expect(getHeightStyles()).toMatchSnapshot();
	});

	test("handles responsive value", async () => {
		expect(getHeightStyles({ s: "50px", l: "50%" })).toMatchSnapshot();
	});
});
