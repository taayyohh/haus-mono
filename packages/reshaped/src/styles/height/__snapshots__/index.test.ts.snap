// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Styles/Height > handles % value 1`] = `
{
  "classNames": [
    "_literal_e6611b",
  ],
  "variables": {
    "--rs-h-s": "50%",
  },
}
`;

exports[`Styles/Height > handles px value 1`] = `
{
  "classNames": [
    "_literal_e6611b",
  ],
  "variables": {
    "--rs-h-s": "50px",
  },
}
`;

exports[`Styles/Height > handles responsive value 1`] = `
{
  "classNames": [
    "_literal_e6611b",
    "_literal--l_e6611b",
  ],
  "variables": {
    "--rs-h-l": "50%",
    "--rs-h-s": "50px",
  },
}
`;

exports[`Styles/Height > handles undefined value 1`] = `null`;

exports[`Styles/Height > handles unit value 1`] = `
{
  "classNames": [
    "_unit_e6611b",
  ],
  "variables": {
    "--rs-h-s": 5,
  },
}
`;
